fe/fe-core/src/main/java/com/starrocks/privilege/StorageVolumePEntryObject.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.privilege;
+
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.storagevolume.StorageVolume;
+
+import java.util.List;
+import java.util.Objects;
+
+public class StorageVolumePEntryObject implements PEntryObject {
+    @SerializedName(value = "id")
+    private String id;
+
+    protected StorageVolumePEntryObject(String id) {
+        this.id = id;
+    }
+
+    public static PEntryObject generate(GlobalStateMgr mgr, List<String> tokens) throws PrivilegeException {
+        if (tokens.size() != 1) {
+            throw new PrivilegeException("invalid object tokens, should have one: " + tokens);
+        }
+        String name = tokens.get(0);
+        if (name.equals("*")) {
+            return new StorageVolumePEntryObject(null);
+        } else {
+            StorageVolume sv = null;
+            try {
+                sv = mgr.getStorageVolumeMgr().getStorageVolumeByName(name);
+            } catch (AnalysisException e) {
```

Comment:
why getStorageVolumeByName throws `AnalysisException`? is this reasonable?

Reply:
I will refactor the exception thrown in another PR.